CMake error: Unable to specify include directories for imported targets

When trying to compile on my home PC (Ubuntu 18.04), I encountered the following error when performing `cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Debug -DDEVEL_TOOLS=ON -DBUILD_TESTS=ON -DBUILD_REFERENCE=ON -DBUILD_OMP=ON -DBUILD_CUDA=ON ../..` on a clean build folder:
```
CMake Error at cmake/package_helpers.cmake:68 (target_include_directories):
  Cannot specify include directories for imported target "gtest".
Call Stack (most recent call first):
  third_party/gtest/CMakeLists.txt:6 (ginkgo_add_external_target)


CMake Error at cmake/package_helpers.cmake:68 (target_include_directories):
  Cannot specify include directories for imported target "gtest_main".
Call Stack (most recent call first):
  third_party/gtest/CMakeLists.txt:8 (ginkgo_add_external_target)


CMake Error at third_party/gtest/CMakeLists.txt:10 (add_library):
  add_library cannot create ALIAS target "GTest::GTest" because target
  "gtest" is IMPORTED.


CMake Error at third_party/gtest/CMakeLists.txt:11 (add_library):
  add_library cannot create ALIAS target "GTest::Main" because target
  "gtest_main" is IMPORTED.
```
Here is the full output: [cmake_output.txt](https://github.com/ginkgo-project/ginkgo/files/2884892/cmake_output.txt)

My CMake version is `3.10.2`.

After some digging, `target_include_directories` can be used with `IMPORTED` libraries since CMake `3.11.0` (here the [MergeRequest from cmake](https://gitlab.kitware.com/cmake/cmake/merge_requests/1264)).

I think we should either change that, or, increase the required version further to `3.11`.

Using`SET_PROPERTY(TARGET ${name} APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${${name}_INCLUDE_DIR})`
instead of `target_include_directories` and
dropping the `add_library(<namespace_name> ALIAS <name>)` (while replacing the `namespace_name` with the `name` given to the macro `ginkgo_add_external_target()`) works for not getting a CMake error.

However, when compiling, I get
```bash
make[2]: *** No rule to make target 'third_party/gflags/build/lib/./libgflags.a', needed by 'benchmark/matrix_generator/matrix_generator'.  Stop.
```
Guess I still need to add something in order for it to work.

--- a/cmake/package_helpers.cmake
+++ b/cmake/package_helpers.cmake
@@ -29,7 +29,8 @@ endfunction()
 #   Create a new target and declare it as `IMPORTED` for libraries or `INTERFACE`
 #       for header only projects.
 #
-#   \param name             Name of external project
+#   \param new_target       New target for the external project
+#   \param external_name    Name of the external project
 #   \param includedir       Path to include directory
 #   \param libdir           Path to library directory
 #   \param build_type       Build type {STATIC, SHARED}
@@ -37,35 +38,35 @@ endfunction()
 #   \param external         Name of the external target
 #   \param header_only      Boolean indicating if this should be a header only target
 #
-macro(ginkgo_add_external_target name includedir libdir build_type debug_postfix external header_only)
+macro(ginkgo_add_external_target new_target external_name includedir libdir build_type debug_postfix external header_only)
     # Declare include directories and library files
-    set(${name}_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/${libdir})
-    set(${name}_INCLUDE_DIR "${CMAKE_CURRENT_BINARY_DIR}/${includedir}")
-    set(${name}_LIBRARY_RELEASE "${${name}_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${CMAKE_${build_type}_LIBRARY_PREFIX}${name}${CMAKE_${build_type}_LIBRARY_SUFFIX}")
-    set(${name}_LIBRARY_DEBUG "${${name}_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${CMAKE_${build_type}_LIBRARY_PREFIX}${name}${debug_postfix}${CMAKE_${build_type}_LIBRARY_SUFFIX}")
+    set(${external_name}_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/${libdir})
+    set(${external_name}_INCLUDE_DIR "${CMAKE_CURRENT_BINARY_DIR}/${includedir}")
+    set(${external_name}_LIBRARY_RELEASE "${${external_name}_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${CMAKE_${build_type}_LIBRARY_PREFIX}${external_name}${CMAKE_${build_type}_LIBRARY_SUFFIX}")
+    set(${external_name}_LIBRARY_DEBUG "${${external_name}_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${CMAKE_${build_type}_LIBRARY_PREFIX}${external_name}${debug_postfix}${CMAKE_${build_type}_LIBRARY_SUFFIX}")
 
     # Create an IMPORTED external library available in the GLOBAL scope
     if (${header_only})
-        add_library(${name} INTERFACE)
+        add_library(${new_target} INTERFACE)
     else()
-        add_library(${name} ${build_type} IMPORTED GLOBAL)
+        add_library(${new_target} ${build_type} IMPORTED GLOBAL)
     endif()
 
     # Set a dependency to the external target (ExternalProject fetcher and builder)
-    add_dependencies(${name} ${external})
+    add_dependencies(${new_target} ${external})
 
     # Set the target's properties, namely library file and include directory
     if (NOT ${header_only})
-        set_target_properties(${name} PROPERTIES IMPORTED_LOCATION_RELEASE ${${name}_LIBRARY_RELEASE})
-        set_target_properties(${name} PROPERTIES IMPORTED_LOCATION_DEBUG ${${name}_LIBRARY_DEBUG})
+        set_target_properties(${new_target} PROPERTIES IMPORTED_LOCATION_RELEASE ${${external_name}_LIBRARY_RELEASE})
+        set_target_properties(${new_target} PROPERTIES IMPORTED_LOCATION_DEBUG ${${external_name}_LIBRARY_DEBUG})
         # Since we do not really manage other build types, let's globally use the DEBUG symbols
         if (NOT CMAKE_BUILD_TYPE MATCHES "[Rr][Ee][Ll][Ee][Aa][Ss][Ee]"
             AND NOT CMAKE_BUILD_TYPE MATCHES "[Dd][Ee][Bb][Uu][Gg]")
-            set_target_properties(${name} PROPERTIES IMPORTED_LOCATION
-                ${${name}_LIBRARY_DEBUG})
+            set_target_properties(${new_target} PROPERTIES IMPORTED_LOCATION
+                ${${external_name}_LIBRARY_DEBUG})
         endif()
     endif()
-    target_include_directories(${name} INTERFACE ${${name}_INCLUDE_DIR})
+    set_target_properties(${new_target} PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${${external_name}_INCLUDE_DIR})
 endmacro(ginkgo_add_external_target)
 
 

